src/ir/ir_context.h

```diff
@@ -51,7 +51,12 @@ class IRContext {
     return operators_.IsRegisteredNode(operators_name);
   }
 
-    // Register an operator and return the stored operator instances.
+  const Operator &GetOrCreateOperator(absl::string_view operator_name,
```

Comment:
hmm.. `Operator` and `Storage` should be created in one place. e.g., declaration of storage or declaration of an operator. I am not sure where you are planning to use this, but this API can lead to silent errors that would become hard to track.  

It looks like you are passing in a constructed `Operator` or `Storage` anyway. Why not use `RegisterOperator` and `GetOperator`? What advantages do you get with a `GetOrCreateOperator` and other variants?

Reply:
You might be right about `Operator`; we know all of the `Operator`s that we want to use for the SQL MVP in advance, and so we can just create them.

For `Storage`s, which are used for columns currently (and will most likely be used for tables in the future) something like `GetOrCreate` looks much more appealing because we do not know in advance the columns that a user will request. Therefore, it is appealing to be able to `GetOrCreate` a `Storage` in a single call when a column reference is being deserialized.

Perhaps the way to solve this is to keep `GetOrCreateNode`, but eliminate `GetOrCreateOperator`, thus enabling it for `Storage`s and disabling it for `Operator`s.

Reply:
Done.